src/utils/arrays.ts

```diff
@@ -45,3 +45,63 @@ export function arrayDiff<T>(a: T[], b: T[]): { added: T[], removed: T[] } {
         removed: a.filter(i => !b.includes(i)),
     };
 }
+
+/**
+ * Helper functions to perform LINQ-like queries on arrays.
+ */
+export class ArrayUtil<T> {
```

Comment:
`Util` sounds like a module of static functions rather than a class to me... Not sure what to suggest at the moment though.

Reply:
Possibly LINQArray or Array2 to fully embrace .net bad practices. Or dump these on the Array prototype and forget that they're custom to the project

Reply:
Yikes. 😨 Let's just leave it as-is for now, can change if we're inspired later.